### ISSUE TITLE
Memory Error with Python 3.8.3 on big TAR-file, no error with Python 3.6.5 

### ISSUE THREAD

**Issue by realSnoopy** [RealityNet/kobackupdec#29]
**Describe the bug**
Memory Error with Python 3.8.3
Python 3.6.5 IS working

it occurs when working on a big single tar file, size in bytes is 2.890.546.688
as mentioned, Phyton 3.6.5 is working without this problem and the tar can be sucessfully decrypted

**Required info (please complete the following information):**
 - Huawei Kobackup version: unknown
 - Host: Windows 10 up2date
 - Kobackup script version: '20200611'
 - Kobackup output log (use -vvv)

using with option -e -> ERROR
without -e -> ERROR 

```
Traceback (most recent call last):
  File "kobackupdec.py", line 1013, in <module>
    main(user_password, backup_path, dest_path, args.expandtar, args.writable)
  File "kobackupdec.py", line 941, in main
    decrypt_backup(password, files_folder, dest_path_out, expandtar)
  File "kobackupdec.py", line 885, in decrypt_backup
    decrypt_files_in_root(decrypt_info, path_in, path_out, expandtar)
  File "kobackupdec.py", line 768, in decrypt_files_in_root
    for x in decrypt_large_entry(decrypt_info, entry,
  File "kobackupdec.py", line 703, in decrypt_large_entry
    decrypt_material, entry.read_bytes()):
  File "D:\System\Python38\lib\pathlib.py", line 1226, in read_bytes
    return f.read()
MemoryError
```

dont sure whether it has really to do with your script or changes made in pathlib library or anything else, on a quick look in both pathlib.py read_bytes() uses open() and read() and the only change i found is in open()

```
return io.open(self, mode, buffering, encoding, errors, newline, opener=self._opener)
return io.open(str(self), mode, buffering, encoding, errors, newline, opener=self._opener)
```

but this can not be the problem...

**Comment by dfirfpi** [comment]
Hi @realSnoopy ,
thanks for reporting. For sure something changed, as you pointed out but, regardless of any change, the code is wrong... I should not read (supposedly) all the input bytes... I'll change that.

**Comment by Mohsyn** [comment]
@realSnoopy did you get it to work? what changes did you make ?

**Comment by Mohsyn** [comment]
@dfirfpi 

I tried running it on py27 - Didn't work   [ surprised on how it worked for Snoopy  - maybe some libs missing]
Tried modifying the read_bytes - Didn't work [ thanks to lack on my expertise in python ]

Would appreciate if you can look into it or guide me on where to make the changes (which function).

Cheers

**Comment by dfirfpi** [comment]
Should be fixed with version 20200705.
